### ISSUE TITLE
no-unexpected-multiline 

### ISSUE THREAD

**Issue by terencechow** [eslint/eslint#8579]
**Tell us about your environment**

* **ESLint Version: 3.16.1**
* **Node Version: 7.5**
* **npm Version: 4.1.2**

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**

```
{
  "parser": "babel-eslint",
  "extends": [
    "standard"
  ],
  "plugins": [
    "babel",
    "promise"
  ],
  "rules": {
    "key-spacing"          : 0,
    "no-useless-escape"    : 0,
    "jsx-quotes"           : [2, "prefer-single"],
    "max-len"              : [2, 120, 2],
    "object-curly-spacing" : [2, "always"]
  }
}

```

**What did you do? Please include the actual source code causing the issue.**

Wrote normal valid es6 code and got an Eslint error. See last question where i include all the details.

**What did you expect to happen?**

No eslint warning /error

**What actually happened? Please include the actual, raw output from ESLint.**

Eslint Error  msg says `Unexpected newline between object and [ of property access. (no-unexpected-multiline)` 

This occurs when using es6 features like destructuring to grab multiple things from an array. If you aren't using `let` / `constant` in front of the array then you get this eslint error as it seems eslint thinks you are using an array. Examples below:

Below produces eslint error
```
let varOne, varTwo
const a = 'abc'

[ varOne, varTwo ] = someFuncThatReturnsArrayOfTwo() // <-- produces error

```

Below would not cause errors because `const` is used so eslint realizes its not accessing an array

```
const a = 'abc'

const [ varOne, varTwo ] = someFuncThatReturnsArrayOfTwo() // <-- does not produce error
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is working as designed -- your code isn't doing what you think it's doing, so ESLint is trying to warn you about that.

`[ varOne, varTwo ]` is parsed as a computed property access on the string `'abc'`, not a separate statement. If you run the code, you'll find that the variable `a` is assigned to an array with two elements, and `varOne` and `varTwo` are still undefined afterwards. This is one of the pitfalls of not using semicolons to separate statements.

To fix this, you can insert a semicolon between the two statements:

```js
let varOne, varTwo
const a = 'abc'

;[ varOne, varTwo ] = someFuncThatReturnsArrayOfTwo() // <-- produces error
```

**Comment by terencechow** [comment]
Very interesting! Thank you for your response. I used const whenever I got
that error but glad to understand the difference. Ty

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
